packages/playwright-core/src/server/trace/recorder/snapshotter.ts

```diff
@@ -105,8 +105,13 @@ export class Snapshotter {
   }
 
   async captureSnapshot(page: Page, callId: string, snapshotName: string, element?: ElementHandle): Promise<void> {
+
+    const noscript_remove = this._context._options.javaScriptEnabled
+      ? `window["${this._snapshotStreamer}"]._removeNoScript = true;`
+      : `window["${this._snapshotStreamer}"]._removeNoScript = false;`;
+
     // Prepare expression synchronously.
-    const expression = `window["${this._snapshotStreamer}"].captureSnapshot(${JSON.stringify(snapshotName)})`;
+    const expression = `${noscript_remove} window["${this._snapshotStreamer}"].captureSnapshot(${JSON.stringify(snapshotName)})`;
```

Comment:
Let's add a test for this functionality in the trace. Take a look at this one for expiration: https://github.com/microsoft/playwright/blob/main/tests/library/trace-viewer.spec.ts#L656-L668. Basically, configure context, perform a `goto` action, and check that text from the `noscript` tag is visible in the snapshot.

You will need `test.use({ javaScriptEnabled: false })` similarly to the test in this PR. Let me know if you have any troubles with writing a test, or any other questions.

Reply:
I think, it should be already covered by the included tests.

But the correctness of the pure visual text representation on screen is hard to prove.
That's why we should look for a consistent behavior of display and text-selection fixes.